### PR TITLE
cli: Add -config flag to "acl authmethod update/create"

### LINKED ISSUE #7761: Optionally configure Auth Methods via the CLI via json blobs
_(Note there is only one auth method type right now: `kubernetes`)_

Currently when you use `consul acl auth-method <create|update>` you pass all of the configuration settings for the auth methods through via command line flags:

```
Usage: consul acl auth-method create -name NAME -type TYPE [options]
...(snip)...

  -kubernetes-ca-cert=<string>
     PEM encoded CA cert for use by the TLS client used to talk with the
     Kubernetes API. May be prefixed with '@' to indicate that the value
     is a file path to load the cert from. This flag is required for
     type=kubernetes.

  -kubernetes-host=<string>
     Address of the Kubernetes API server. This flag is required for
     type=kubernetes.

  -kubernetes-service-account-jwt=<string>
     A Kubernetes service account JWT used to access the TokenReview
     API to validate other JWTs during login. This flag is required for
     type=kubernetes.
```

These are passed through [the API](https://www.consul.io/api/acl/auth-methods.html#config) as a json-serialized `map[string]interface{}`. Once on the server they are lazily decoded using `mapstructure` into proper structs.

This setup followed work similarly done with the CA provider config being semi-generic to support possibly making these things be plugins in the future.

Now that we're adding more auth methods we have to also make the CLI CRUD subcommands work with them, and it's not looking like adding all of the config settings as flags is going to be reasonable from a UX (and quality of life) perspective since they have ~15 new options.

The proposal is to add an alternate way to provide the auth method configuration other than straight flags. That alternate way would function similarly to `consul config write -` and `consul acl policy create ... -rules=<RAW | - | @FILENAME>`.

The user could say `consul acl auth-method create ... -config=@method.json` or specific it inline or via stdin. The provided structure would need to be JSON for this iteration as we do not have any current need in the plumbing currently to support an HCL representation (which should not rule it out in the future, just for MVP).

The CLI would just take the input and encode it into `map[string]interface{}` and do minimal to no CLI-side validation of the payload before firing it off to the consul API. The API will do validation already so there's no need for the CLI to replicate all of that other than stuff in the envelope around `Config` like name/type/description.

Similar changes would have to happen to `consul acl auth-method update`. That currently has mechanisms for doing "append" semantics or "replace" semantics when doing an update and for this issue I believe when bringing your own complete config blob we should exclusively do the "replace" semantics.

### PR DESCRIPTION
**Pull request by s-christoff**
Fixes #7761

### CODE REVIEW

File: command/acl/authmethod/update/authmethod_update.go [hunk 1]
```diff
@@ -64,6 +66,23 @@ func (c *cmd) init() {
 		"",
 		"A description of the auth method.",
 	)
+	c.flags.StringVar(
```
Comment (rboyer) [inline command/acl/authmethod/update/authmethod_update.go:69]: You duplicated this in the move and I think the first one is the one you didn't want.

File: command/acl/authmethod/update/authmethod_update.go [hunk 2]
```diff
@@ -64,6 +66,23 @@ func (c *cmd) init() {
...
+			"to indicate that the value is a file path to load the rules from. '-' may also be "+
+			"given to indicate that the rules are available on stdin",
+	)
+
+	c.flags.StringVar(
+		&c.config,
+		"config",
+		"",
+		"The configuration for the auth method. Must be JSON. The config is updated as one field"+
+			"May be prefixed with '@' to indicate that the value is a file path to load the rules from. "+
```
Comment (rboyer) [inline command/acl/authmethod/update/authmethod_update.go:83]:
```suggestion
			"May be prefixed with '@' to indicate that the value is a file path to load the config from. "+
```

File: command/acl/authmethod/update/authmethod_update.go [hunk 3]
```diff
@@ -64,6 +66,23 @@ func (c *cmd) init() {
...
+			"'-' may also be given to indicate that the rules are available on stdin. ",
```
Comment (rboyer) [inline command/acl/authmethod/update/authmethod_update.go:84]:
```suggestion
			"'-' may also be given to indicate that the config are available on stdin. ",
```

File: command/acl/authmethod/update/authmethod_update.go [hunk 4]
```diff
@@ -91,7 +110,7 @@ func (c *cmd) init() {
 
 	c.flags.BoolVar(&c.noMerge, "no-merge", false, "Do not merge the current auth method "+
 		"information with what is provided to the command. Instead overwrite all fields "+
-		"with the exception of the name which is immutable.")
+		"with the exception of the name which is immutable. Cannot be used with '-config-file'")
```
Comment (rboyer) [inline command/acl/authmethod/update/authmethod_update.go:113]: I thought you already removed this?

File: command/acl/authmethod/update/authmethod_update.go [hunk 5]
```diff
@@ -171,10 +204,27 @@ func (c *cmd) Run(args []string) int {
 	} else {
 		methodCopy := *currentAuthMethod
 		method = &methodCopy
-
 		if c.description != "" {
 			method.Description = c.description
 		}
+		if c.displayName != "" {
```
Comment (rboyer) [inline command/acl/authmethod/update/authmethod_update.go:210]: I didn't merge display name yet.

Comment (s-christoff) [reply command/acl/authmethod/update/authmethod_update.go:210]: I aM frOm tHe fUtUrE

**Pull request merged**